flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectSinkOperator.java

```diff
@@ -51,9 +51,9 @@ public void handleOperatorEvent(OperatorEvent evt) {
     }
 
     @Override
-    public void finish() throws Exception {
+    public void close() throws Exception {
         sinkFunction.accumulateFinalResults();
-        super.finish();
+        super.close();
     }
```

Comment:
Why did we need to change this? Does this impact any other connectors?

Reply:
The change has been originally contributed by @AHeise. I haven't evaluated this in depth, however the original commit message suggests:

> DataStream#executeAndCollect expects the CollectSinkOperator to register the accumulator at the end of the application or fails with some exception.
However, a stop-with-savepoint without drain would not trigger CollectSinkOperator#finish and thus skip the registration.